### PR TITLE
fix(for-of): Skip logic on small scrolls

### LINKED ISSUE #7184: IgxCombo - Item selection after scrolling the combo does not work correctly
## Description  
When clicking on a combo item after the virtualized container has been scrolled, selection behaves inconsistently  
 * igniteui-angular version: 9.1x
 * browser: N/A

## Steps to reproduce  

1. Run the dev demos and open the combo sample (http://localhost:4200/combo)
2. Open the "State" combo
3. Scroll the items container to the bottom
4. Scroll back to the top
5. Click on the first item ("Michigan")

## Result  
"Michigan" is **not** selected.
If you click again, it is selected. If you click on **another** state (ie "Indiana"), "Michigan" is **deselected**

## Expected result  
"Michigan" is selected

## Notes
Clicks work fine before scrolling the container. Works fine with keyboard navigation.

### LINKED ISSUE #7202: IgxForOf - Virtualized items do not properly trigger change detection
## Description  
After scrolling a virtualized container created w/ `IgxForOf`, user input on the items does not trigger change detection as expected

 * igniteui-angular version: 9.1.0-alpha.7
 * browser: N/A

## Steps to reproduce  

1. Open this [StackBlitz sample](https://stackblitz.com/edit/igx-for-of-910-alpha?file=src%2Fapp%2FigxFor%2FigxFor.component.ts)
2. Click on an item in the list - it's contents is populated in the input and the item has a gray background
3. Scroll the list
4. Click on an item

## Result  
The item is not marked as selected

## Expected result  
The item is marked as selected

## Notes
Manually calling `ChangeDetectorRef.detectChanges()` in the click handler causes proper behavior

### LINKED ISSUE #7206: Memory consumption/leak issues in Grid.
## Description  
Memory consumption rises when scrolling (with small scrolls) in Grid.

Probably caused by: https://github.com/IgniteUI/igniteui-angular/pull/6910/files?file-filters%5B%5D=.ts&hide-deleted-files=true#diff-d637aea7875639647b3e0e557ff6d1fbR889

**Note: The performance optimization was implemented to resolve issue - https://github.com/IgniteUI/igniteui-angular/issues/3083
Any fix should be tested to ensure that is not regressed**

 * igniteui-angular version: 9.1.x
 * browser: Any

## Steps to reproduce :
1. Open a sample with a grid that has a scrollbar.
2. Do a memory profile of the current state.
3. Scroll up/down with small scroll increments (drag the scrollbar slowly up/down, use mouse wheel etc.)
4. Do another memory profile and note that the number of objects rendered by the igxForOf have increased. Number of components like igxGridRow/igxTemplateOutlet have increased and remain in memory.

It appears that the detached views remain in memory, hence ViewChildren collections like grid.tmpOutlets contains reference to both the rendered and detached outlets. 
On destroy, we have some custom logic that cleans all template outlets (in the grid's ngOnDestroy) so igxGridRow instances should be destroyed, however the igxTemplateOutlet instances that are currently detached will not be destroyed (ngOnDestroy never fires for them).

## Result  
Memory consumption increases the more you scroll and on destroy of the grid not all inner components are destroyed.

## Expected result  
No memory consumption increase and no memory leaks.

### PR DESCRIPTION
**Pull request by rkaraivanov**
Closes #7206
Closes #7202 
Closes #7184 

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by ChronosSF** [comment]
This should also close #7202

**Comment by ddincheva** [comment]
> This should also close #7202 and #7184

I managed to do some testing and the described issues are resolved with this fix.

**Comment by zdrawku** [comment]
I've also tested it with multiple mobile (touch) devices under safari and chrome - the scrolling is smooth and there are no problems with the intertion scrolling. Related to https://github.com/IgniteUI/igniteui-angular/issues/3083

File: projects/igniteui-angular/src/lib/grids/selection/selection.service.ts [hunk 1]
```diff
@@ -161,6 +161,7 @@ export class IgxGridCRUDService {
 
     begin(cell): void {
         const newCell = this.createCell(cell);
+        cell.focused = true;
```
Comment (rkaraivanov) [inline projects/igniteui-angular/src/lib/grids/selection/selection.service.ts:164]: Why is this here? The `cell` in this context is not the IgxGridCellComponent. Moreover it does not have a focused property at all.

File: projects/igniteui-angular/src/lib/grids/cell.component.html [hunk 2]
```diff
@@ -11,21 +11,21 @@
...
     <ng-container *ngIf="column.dataType === 'string'">
         <igx-input-group displayDensity="compact">
-            <input igxInput [value]="editValue" (input)="editValue = $event.target.value" [igxFocus]="true" />
+            <input igxInput [value]="editValue" (input)="editValue = $event.target.value" [igxFocus]="focused" />
         </igx-input-group>
     </ng-container>
     <ng-container *ngIf="column.dataType === 'number'">
         <igx-input-group displayDensity="compact">
-            <input igxInput [value]="editValue" (input)="editValue = $event.target.value" [igxFocus]="true" type="number">
+            <input igxInput [value]="editValue" (input)="editValue = $event.target.value" [igxFocus]="focused" type="number">
```
Comment (rkaraivanov) [inline projects/igniteui-angular/src/lib/grids/cell.component.html:19]: Maybe bind to `active` instead?

Comment (ddincheva) [reply projects/igniteui-angular/src/lib/grids/cell.component.html:19]: we need to move focus to the edit template only initially, the template should be blurred if we perform scroll, but the cell will remain active, so if we bind the igxFocus to active property .. this will retrigger focus over the template again and again on each change detection  which broke the scrolling experience in the Igx-HierarchicalGrid.
